doc/api_assets/api.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+// Check if we have JavaScript support
+document.querySelector(':root').classList.add('has-js');
+
+// Restore user mode preferences
+const kCustomPreference = 'customDarkTheme';
+const userSettings = sessionStorage.getItem(kCustomPreference);
+const themeToggleButton = document.getElementById('theme-toggle-btn');
+if (userSettings === null && window.matchMedia) {
+  const mq = window.matchMedia('(prefers-color-scheme: dark)');
+  if ('onchange' in mq) {
+    function mqChangeListener(e) {
+      document.documentElement.classList.toggle('dark-mode', e.matches);
+    }
+    mq.addEventListener('change', mqChangeListener);
+    if (themeToggleButton) {
+      themeToggleButton.addEventListener('click', function() {
+        mq.removeEventListener('change', mqChangeListener);
+      }, { once: true });
+    }
+  }
+  if (mq.matches) {
+    document.documentElement.classList.add('dark-mode');
+  }
+} else if (userSettings === 'true') {
+  document.documentElement.classList.add('dark-mode');
+}
+if (themeToggleButton) {
+  themeToggleButton.hidden = false;
+  themeToggleButton.addEventListener('click', function() {
+    sessionStorage.setItem(
+      kCustomPreference,
+      document.documentElement.classList.toggle('dark-mode')
+    );
+  });
+}
+
+// Handle pickers with click/taps rather than hovers
+const pickers = document.querySelectorAll('.picker-header');
+for (const picker of pickers) {
+  picker.addEventListener('click', (e) => {
+    if (!e.target.closest('.picker')) {
+      e.preventDefault();
+    }
+
+    if (picker.classList.contains('expanded')) {
+      picker.classList.remove('expanded');
+    } else {
+      for (const other of pickers) {
+        other.classList.remove('expanded');
+      }
+
+      picker.classList.add('expanded');
```

Comment:
A few suggestions:

- We shouldn't add event listener to a list-item, but rather to the link itself for accessibility reasons
- I think it makes sense to add a click handler to the whole document so the menu closes when the user clicks somewhere else.
- I think it makes sense to close the menu when the user presses the ESC key.

```suggestion
const pickers = document.querySelectorAll('.picker-header>a');
const closeAllPickers = () => {
  for (const picker of pickers) {
    picker.parentNode.classList.remove('expanded');
  }
  window.removeEventListener('click', closeAllPickers);
  window.removeEventListener('keydown', onKeyDown);
};
const onKeyDown = (e) => {
  if (e.key === 'Escape') {
    closeAllPickers();
  }
};
for (const picker of pickers) {
  picker.addEventListener('click', (e) => {
    e.preventDefault();
    const { parentNode } = picker;
    const shouldOpen = !parentNode.classList.contains('expanded');
    
    closeAllPickers();
    if(shouldOpen) {
      parentNode.classList.add('expanded');
      requestAnimationFrame(() => {
        window.addEventListener('click', closeAllPickers);
        window.addEventListener('keydown', onKeyDown);
      })
```

Reply:
Thanks for the very good hints. I've incorporated all of them in the refactoring.